pkg/controller/injector/k8s_binding_injector_test.go

```diff
@@ -29,6 +30,123 @@ import (
 	kapi "k8s.io/kubernetes/pkg/api"
 )
 
+func TestCreateSerializedSecret(t *testing.T) {
+	cases := []struct {
+		name   string
+		cred   *brokerapi.Credential
+		secret *v1.Secret
+	}{
+		{
+			name: "string type",
+			cred: &brokerapi.Credential{
+				"Hostname": "host",
+				"Port":     "123",
+				"Username": "user",
+				"Password": "password!@#!@#!0)",
+			},
+			secret: &v1.Secret{
+				Data: map[string][]byte{
+					"Hostname": []byte("host"),
+					"Port":     []byte("123"),
+					"Username": []byte("user"),
+					"Password": []byte("password!@#!@#!0)"),
+				},
+			},
+		},
+		{
+			name: "float type",
+			cred: &brokerapi.Credential{
+				"Hostname": "host",
+				"Port":     "123",
+				"Username": "user",
+				"Password": 1.23,
+			},
+			secret: &v1.Secret{
+				Data: map[string][]byte{
+					"Hostname": []byte("host"),
+					"Port":     []byte("123"),
+					"Username": []byte("user"),
+					"Password": []byte("1.23"),
```

Comment:
> That is why we should test this code.

I'm not at all saying we shouldn't test this code. I'm only arguing for the _right_ tests. Before this PR, the tests in `k8s_binding_injector_test.go` were at the right level. They tested that when injection occurs, the resulting secret has in it all the same data as the credentials object. i.e. It tested that _injection_ works and it does so agnostic of what underlying serialization process is in play. The more complex bit of ensuring that serialization process works was left to its own tests in `serializer_test.go`.

After this PR, we are testing serialization _twice_ and adding no further value because the "other business logic" you say will one day exist in the injector currently doesn't. When the day comes, _yes_, we do need to amend these tests to test that new business logic, but even then, I can't see what benefit arises from duplicate serialization tests. When more business log gets added, the new tests that are added along with it shouldn't focus on serialization, but should focus on the business logic.